File: workshop_content/KI_WORKSHOP/Java_Vergleichsgruppe/src/main/java/provisionierung/ProvisionsRechner.java
package provisionierung;

import provisionierung.domain.*;

/**
 * Aufgabe: Implementiere die Provisionsberechnung basierend auf den Tests und dem Legacy-Code.
 *
 * Analyse:
 * - Die Testfälle definieren das erwartete Verhalten
 * - Der Legacy-Code (siehe Projekt-Root) enthält die originale Berechnungslogik
 *
 * Anforderungen:
 * - Keine UI-Abhängigkeiten
 * - Nutze das ProvisionContext-Interface für zeitabhängige Logik
 * - Alle magischen Zahlen als Konstanten definieren
 * - Modernes Error-Handling
 *
 * Hinweis: Du darfst das Domänenmodell anpassen wenn nötig.
 */
public class ProvisionsRechner {

    // Definiere hier deine Konstanten

    private final ProvisionContext context;

    public ProvisionsRechner() {
        this(new SystemProvisionContext());
    }

    public ProvisionsRechner(ProvisionContext context) {
        this.context = context;
    }

    public ProvisionsErgebnis berechne(ProvisionsEingabe eingabe) {
        throw new UnsupportedOperationException("Implementiere die Berechnungslogik basierend auf den Tests.");
    }
}

File: workshop_content/KI_WORKSHOP/VB.Net_Vergleichsgruppe/Provisionierung/ProvisionsRechner.vb
Imports Provisionierung.Domain

''' <summary>
''' Aufgabe: Implementiere die Provisionsberechnung basierend auf den Tests und dem Legacy-Code.
''' 
''' Analyse:
''' - Die Testfälle definieren das erwartete Verhalten
''' - Der Legacy-Code (siehe Projekt-Root) enthält die originale Berechnungslogik
''' 
''' Anforderungen:
''' - Keine UI-Abhängigkeiten (kein MessageBox, kein DoEvents)
''' - Nutze das IProvisionContext-Interface für zeitabhängige Logik
''' - Alle magischen Zahlen als Konstanten definieren
''' - Modernes Error-Handling statt On Error Resume Next
''' 
''' Hinweis: Du darfst das Domänenmodell anpassen wenn nötig.
''' </summary>
Public Class ProvisionsRechner

    ' Definiere hier deine Konstanten

    Private ReadOnly _context As IProvisionContext

    Public Sub New(Optional context As IProvisionContext = Nothing)
        _context = If(context, New SystemProvisionContext())
    End Sub

    Public Function Berechne(eingabe As ProvisionsEingabe) As ProvisionsErgebnis
        Throw New NotImplementedException("Implementiere die Berechnungslogik basierend auf den Tests.")
    End Function

End Class
